sdk/src/azure/core/az_span.c

```diff
@@ -983,6 +983,33 @@ AZ_NODISCARD AZ_INLINE bool _az_span_url_should_encode(uint8_t c)
   }
 }
 
+AZ_NODISCARD int32_t _az_span_url_encode_calc_length(az_span source)
+{
+  _az_PRECONDITION_VALID_SPAN(source, 0, true);
+
+  int32_t const source_size = az_span_size(source);
+  if (source_size == 0)
+  {
+    return 0;
+  }
+
+  uint8_t* const src_ptr = az_span_ptr(source);
+  int32_t required_symbols_to_be_added = 0;
+  int32_t src_idx = 0;
+  do
+  {
+    uint8_t c = src_ptr[src_idx];
+    if (_az_span_url_should_encode(c))
+    {
+      required_symbols_to_be_added
+          += 2; // Adding '%' plus 2 digits (minus 1 as original symbol is counted as 1)
```

Comment:
nit: Move the comment above the line so the code fits on one line.
```suggestion
      // Adding '%' plus 2 digits (minus 1 as original symbol is counted as 1)
      required_symbols_to_be_added += 2; 
```